doc/conf.py

```diff
@@ -338,9 +338,12 @@ def make_carousel_thumbs(app, exception):
 def filter_search_index(app, exception):
     if exception is not None:
         return
-    print('Removing methods from search index')
-
     searchindex_path = os.path.join(app.builder.outdir, 'searchindex.js')
+
+    if not os.path.exists(searchindex_path):
```

Comment:
Not a sphinx expert, but maybe you could use `app.builder` to only run `filter_search_index` where the builder is the HTML one?

Alternatively, there may be a sphinx-specific to register hooks (or event or whatever the sphinx actual terms is) only for some builders.